Update libmesh; fix VTKIO output in parallel

jwpeterson opened the pull request:
~~Just pushing this to see if it will pass the tests, we probably want to wait awhile until the next libmesh update, at least until @friedmud has a chance to fix up the PetscVector::operator() locking stuff.~~

I'd like to get this merged ASAP.  The libmesh update now includes:
- The fix for C++11 threaded calls to PetscVector::operator().
- Fixes for VTKIO output in parallel.
- C++11 tests for `std::to_string()`.
- Optimizations to `FE`/`FEMap` to speed up `reinit()`.

Refs libMesh/libmesh#882.

jwpeterson committed:
Regold output files for VTK output tests.

This file format now contains the element processor IDs by default,
and correctly outputs the solution in parallel.

Refs libMesh/libmesh#882.

jwpeterson committed:
Update libMesh.

This brings in a fix to the VTKIO class that goes along with the
regolded tests in this PR.

Refs #000.

friedmud commented:
We _do_ need a libMesh update - for one thing this has the fix in it for Dina's restart problem.

I'll try to get that petsc_vector fix in tonight.

jwpeterson commented:
> We do need a libMesh update - for one thing this has the fix in it for Dina's restart problem.

That one was already merged this week!

permcody commented:
Didn't you tell me that MIT students dislike the frequent updates?

On Wed, Mar 30, 2016 at 5:02 PM Derek Gaston <notifications@github.com>
wrote:

> We *do* need a libMesh update - for one thing this has the fix in it for
> Dina's restart problem.
>
> I'll try to get that petsc_vector fix in tonight.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6662#issuecomment-203674580>
>

friedmud commented:
They do: but this issue has been ongoing for nearly a month.  That's not really "frequent" ;-)

Of course: a lot of that time was spent just trying to figure out what the hell the problem was!

jwpeterson committed:
Only print "basic functionality" check when C++11 is enabled.

jwpeterson commented:
Let's merge this on April 14th.  This is also the day @milljm scheduled to stop providing pre-built copies of libMesh on the INL clusters.

jwpeterson commented:
Hold off on merging this a bit longer.  I need to rebase it on top of Roy's latest FE optimization branxh.

jwpeterson commented:
OK, when this finishes testing, I think we're good to do the libmesh update.

jwpeterson commented:
Crap, @roystgnr's FEMap optimizations apparently break several tests in Yak.  The crashes mostly look like segfaults, probably due to not calling `get_xyz()` on enough stuff...  It's the only app that breaks, though, and I'd really like to have these FE optimizations in here for this libmesh update.  Do you think you'll have a chance to look at this?

roystgnr commented:
I took a quick peek to see if it'd be a quick fix, but the only get_xyz() calls I see appear to be in pre-init methods where the prerequesting ought to come out correct.  I'll see if I can duplicate this (in dbg mode so I can make some sense of it), but I only keep the non-open-source INL stuff on my laptop, I haven't rebuilt libMesh or MOOSE there in a while, so it'll take some time.

jwpeterson commented:
OK, if I can get a dbg mode stack trace here, I'll also bring in @snschune's expertise on this end.

friedmud commented:
Before we do an update I would like to verify that Dina's problem is
fixed... she's still reporting that it doesn't work after the last libMesh
update...
On Thu, Apr 14, 2016 at 12:04 PM John W. Peterson <notifications@github.com>
wrote:

> OK, if I can get a dbg mode stack trace here, I'll also bring in @snschune
> <https://github.com/snschune>'s expertise on this end.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6662#issuecomment-210020346>
>

jwpeterson commented:
> Before we do an update I would like to verify that Dina's problem is
> fixed... she's still reporting that it doesn't work after the last libMesh
> update...

I grabbed your dinas_problem repo and ran the test problem.  It prints:

```
peterson(~/projects/dinas_problem) (master)$ ./dprob-opt 
 Mesh Information:
  elem_dimensions()={3}
  spatial_dimension()=3
  n_nodes()=624899
    n_local_nodes()=624899
  n_elem()=4160708
    n_local_elem()=4160708
    n_active_elem()=3647744
  n_subdomains()=1
  n_partitions()=1
  n_processors()=1
  n_threads()=1
  processor_id()=0

Same!
```

Anything else we need to check?  If it doesn't work for her, maybe she's just not up-to-date on libmesh or something...

jwpeterson commented:
![remesh](https://cloud.githubusercontent.com/assets/1775907/14536768/53f30692-0231-11e6-94b5-c9531ea92bd2.png)

Screenshot of "remesh.e" looks sane.

snschune commented:
Not surprising that it fails the NDA tests. The transport sweep uo directly calls low level routines (apparently) in a way/order they are not supposed to be called.

jwpeterson committed:
Pre-request xyz in Assembly::buildFE().

This is needed at least in Yak, where they use the (x,y,z) locations
of the quadrature points of the FE returned by Assembly::getFE().

Refs libmesh/libMesh#892.

friedmud commented:
@jwpeterson I agree - it works for me.  No idea why it's still not working for her.  I'll have to go see what's up.

So - I'm not standing in the way of this update...
